RAC-5619: add SOL/VNC into OVA Post Test

panpan0000 opened the pull request:
Background:
in this PR, OVA-Post-Test  is added with VNC record/SOL logs.
NOTE: [flvrec.py](https://github.com/panpan0000/vnc2flv/blob/master/tools/flvrec.py) is updated to support this.

Test Done:
http://10.**.**.**5:8080/job/MasterCI_VNC_OVA_TEST/35/artifact/OVA_POST_TEST/
![image](https://user-images.githubusercontent.com/14049268/28359764-eb14aa32-6ca5-11e7-8662-14d44e52e5d6.png)



Details in code:

- ```jobs/build_ova/ansible/main.yml```
     -- ```dns_server_ip``` : it's for Jenkins Sandbox , which OVA need DNS to be able to connect to internet, to download flvrec.py from github.
    -- ```BUILD_ID```: fetch Jenkins BUILD_ID env variable, and attach to video record file name.
    --``` nohup bash vnc_record.sh &```: if without nohup, the background script will exist once ansible exist.
- ```jobs/build_ova/ova_post_test.groovy```
    --  typically, ```./build-config``` is invoked in ```FunctionTest.groovy```, but at this moment, we need vnc_record.sh being generated. so ```BMC_VNODE_CREDS``` and ```build-config``` is required earlier than before 


- ```test.sh.in```
    -- previously, once FIT fail, the ```fetchOVAlog``` will never be invoked .  so I used signal handler(```trap```) to ensure it's run.

panpan0000 commented on jobs/build_ova/ansible/main.yml:
```diff
@@ -46,21 +49,111 @@
+    - name: Stop VNC Record, sleep 2 sec to ensure FLV finishes the disk I/O before VM destroyed
+      become: yes
+      shell: |
+          kill -9 $(ps -lef|grep flvrec.py|grep -v grep| awk '{print $4}')
```
why not ```pkill -f``` ?
see below.
https://stackoverflow.com/questions/37042643/ansible-shell-task-with-pgrep-keeps-failing-works-in-normal-shell

panpan0000 commented on jobs/build_ova/ansible/main.yml:
```diff
@@ -46,21 +49,111 @@
+        if [ ! -z $BUILD_ID ]; then
+             export fname_prefix=${fname_prefix}_b${BUILD_ID}
+        fi
+        nohup bash vnc_record.sh /var/log/rackhd $fname_prefix > vnc_record.log   &
```
using nohup . otherwise, the background running script will exit once ansible exit.

PengTian0 commented:
Can we download flvrec.py in local host, then copy it to ova just like generate-sol-log.sh?
In this way, the vnc_record.sh doesn't need to access public network.

panpan0000 commented:
Hi, @PengTian0 , I just fixed what @changev  missed before. when he added the gateway, it still can not connect with internet because DNS is missing.
so I added it back.
for this time, I would like to  ```pip install vnc2flv``` and download ```flvrec.py```.
next time, there may be other CI test setup requires OVA to access internet during ova-post-test period.
So I think it's fair and make sense to make OVA has the option to connect with internet during ova-post-test.

and forget to mention, the DNS is not usable for Jenkins sandbox, no necessary for Production Jenkins

PengTian0 commented:
:+1:
